syft/frameworks/torch/tensors/interpreters/large_precision.py

```diff
@@ -95,51 +102,57 @@ def __iadd__(self, other):
+        )
+        b = LargePrecisionTensor._internal_representation_to_large_ints(
+            other, internal_precision[self.internal_type]
+        )
+        # We need to divide the result of the multiplication by the precision
+        return self._create_tensor_from_numpy((a * b) / (self.base ** self.precision_fractional))
+
+    __mul__ = mul
+
     def fix_large_precision(self):
-        self.child = self._create_internal_tensor()
+        self.child = self._create_internal_representation()
         return self
 
     def float_precision(self):
         """
-        Restore the tensor expressed now as a matrix for each original item.
+        Restore the tensor from the internal representation.
 
         Returns:
             tensor: the original tensor.
         """
-        # We need to pass the PyTorch tensor to Numpy to allow the intermediate large number.
-        # An alternative would be to iterate through the PyTorch tensor and apply the restore function.
-        # This however wouldn't save us from creating a new tensor
-        ndarray = self.child.numpy()
-        result = self._restore_tensor_into_numbers(ndarray, internal_precision[self.internal_type])
-        return torch.from_numpy(result.reshape(ndarray.shape[:-1]).astype(np.float32))
+        result = self._internal_representation_to_large_ints(
+            self.child, internal_precision[self.internal_type]
+        )
+        result /= self.base ** self.precision_fractional
         # At this point the value is an object type. Force cast to float before creating torch.tensor
+        return torch.from_numpy(result.reshape(self.child.shape[:-1]).astype(np.float32))
 
-    def _restore_tensor_into_numbers(self, number_array, precision):
-        """Creates an numpy array containing the original numbers"""
-        number_array = number_array.reshape(-1, number_array.shape[-1])
-        result = []
-        for elem in number_array:
-            result.append(self._restore_number(elem, precision))
-        return np.array(result)
-
-    @staticmethod
-    def _adjust_to_shape(to_adjust, shape, fill_value=0) -> torch.Tensor:
-        """Numbers with the same precision can be represented with different matrices.
-        This function adjusts the shapes of two tensors to be the same and fill the empty cells with the
-        provided `fill_value`.
-        We assume only the last dimension needs to be adjusted.
-        Original input tensors should have the same dimensions.
-        """
-        diff_dim = shape[-1] - to_adjust.shape[-1]
-        new_shape = to_adjust.shape[:-1] + (diff_dim,)
-
-        if not fill_value:
-            filler = torch.zeros(size=new_shape, dtype=to_adjust.dtype)
-        else:
-            filler = torch.ones(size=new_shape, dtype=to_adjust.dtype)
-        return torch.cat([filler, to_adjust], len(shape) - 1)
-
-    def _split_number(self, number, bits):
+    def _split_number(self, number, bits) -> list:
```

Comment:
Also make this one static and ask that internal type be provided in the signature if needed